spec/requests/admin/articles_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'rails_helper'
+
+RSpec.describe "Admin::Articles", type: :request do
+  context "while signed in as a super admin" do
+    before do
+      sign_in(@super_admin_no_org)
+    end
+
+    describe "GET #new" do
+      it "lets the user load the new article page" do
+        get new_admin_article_path
+        expect(response).to be_successful
+      end
+    end
+
+    describe "POST #create" do
+      it "lets the user create articles" do
+        post admin_articles_path(article: attributes_for(:article))
+        expect(response).to be_redirect
+        expect(Article.all.count).to eq 1
+      end
+
+      it "article questions can't be blank" do
+        post admin_articles_path(article: attributes_for(:article).merge(question: ""))
+        expect(subject).to render_template("new")
+        expect(flash[:error]).to be_present
+        expect(Article.all.count).to eq 0
+      end
+
+      it "article answers can't be blank" do
+        post admin_articles_path(article: attributes_for(:article).merge(content: ""))
+        expect(subject).to render_template("new")
+        expect(flash[:error]).to be_present
+        expect(Article.all.count).to eq 0
+      end
+
+      it "article for_banks and for_partners attributes can't both be false" do
+        post admin_articles_path(article: attributes_for(:article).merge(
+          for_banks: false,
+          for_partners: false
+        ))
+        expect(subject).to render_template("new")
+        expect(flash[:error]).to be_present
+      end
+    end
+
+    describe "GET #edit" do
+      it "lets the user load the edit article page" do
+        article = create(:article)
+        get edit_admin_article_path(article)
+        expect(response).to be_successful
+      end
+    end
+
+    describe "PATCH #update" do
+      it "lets the user update articles" do
+        article = create(:article)
+        patch admin_article_path(
+          {
+            id: article.id,
+            article: {
+              question: "updated question",
+              for_banks: false,
+              for_partners: true
+            }
+          }
+        )
+        expect(response).to be_redirect
+        article.reload
+        expect(article.question).to eq "updated question"
+        expect(article.for_banks).to eq false
+        expect(article.for_partners).to eq true
+      end
+
+      it "article questions can't be blank" do
+        article = create(:article)
+        patch admin_article_path(
+          {
+            id: article.id,
+            article: {
+              question: ""
+            }
+          }
+        )
+        expect(subject).to render_template("edit")
+        expect(flash[:error]).to be_present
+        article.reload
+        expect(article.question).to eq "question"
+      end
+
+      it "article answers can't be blank" do
```

Comment:
Aren't these actually testing model validations? They shouldn't need request specs.

Reply:
I'll move the validation tests to a model spec.

Reply:
Added a comment on this in the model spec. 

Validations are better off in the model spec than in here because validations aren't just important to flows like you testing here but also in other contexts. Say, for example, you added a script/job to create `Questions` and you'll want to be sure that the script/job also can't create Questions without answers as well. Instead of adding specs to both areas of the codebase, it is preferable to control data integrity at the model level and then if that doesn't work at the database level (I've added a comment on this).

Let me know if you have any other questions, feel free to reach out on the RfG slack.

Reply:
I'm in the process of updating the specs to test the Question model, not the Article model. I'll move the validation tests to the model spec. I'll keep this advice in mind in the future. Thanks for the feedback!

Reply:
I made the above mentioned updates.